node/src/accountant/scanners/mod.rs

```diff
@@ -319,23 +319,32 @@ impl PayableScanner {
 
     fn separate_id_triples_by_existent_and_nonexistent_fingerprints<'a>(
         &'a self,
-        sent_payments: &'a [&'a PendingPayable],
+        sent_payables: &'a [&'a PendingPayable],
     ) -> (Vec<PendingPayableTriple>, Vec<PendingPayableTriple>) {
-        let hashes = sent_payments
+        fn join<'a>(
+            ((rowid_opt, hash), pending_payable): ((Option<u64>, H256), &'a &'a PendingPayable),
+        ) -> PendingPayableTriple<'a> {
+            PendingPayableTriple::new(&pending_payable.recipient_wallet, hash, rowid_opt)
+        }
+
+        let hashes = sent_payables
             .iter()
             .map(|pending_payable| pending_payable.hash)
             .collect::<Vec<H256>>();
-        self.pending_payable_dao
+
+        let sent_payables_sorted_by_hashes = sent_payables
+            .iter()
+            .sorted_by(|a, b| Ord::cmp(&a.hash, &b.hash));
+
+        let rowid_pairs_sorted_by_hashes = self
+            .pending_payable_dao
             .fingerprints_rowids(&hashes)
             .into_iter()
-            .zip(sent_payments.iter())
-            .map(
-                |((rowid_opt, hash), pending_payable)| PendingPayableTriple {
-                    recipient: &pending_payable.recipient_wallet,
-                    hash,
-                    rowid_opt,
-                },
-            )
+            .sorted_by(|(_, hash_a), (_, hash_b)| Ord::cmp(&hash_a, &hash_b));
+
+        rowid_pairs_sorted_by_hashes
```

Comment:
Should there be some code in here to verify that the two lists of hashes are identical, and panic if they're not?